applications/StructuralMechanicsApplication/custom_strategies/custom_builder_and_solver/explicit_builder_and_solver.hpp

```diff
@@ -0,0 +1,405 @@
+#define  KRATOS_EXPLICIT_BUILDER_AND_SOLVER
+
+
+/* System includes */
+
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+/* External includes */
+
+
+/* Project includes */
+#include "includes/define.h"
+#include "solving_strategies/builder_and_solvers/builder_and_solver.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+
+/**@name Kratos Globals */
+/*@{ */
+/*@} */
+/**@name Type Definitions */
+/*@{ */
+/*@} */
+/**@name  Enum's */
+/*@{ */
+/*@} */
+/**@name  Functions */
+/*@{ */
+/*@} */
+/**@name Kratos Classes */
+/*@{ */
+
+template<class TSparseSpace,
+         class TDenseSpace, //= DenseSpace<double>,
+         class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+         >
+class ExplicitBuilderAndSolver
+    : public BuilderAndSolver< TSparseSpace, TDenseSpace, TLinearSolver >
+{
+public:
+    /**@name Type Definitions */
+    /*@{ */
+    KRATOS_CLASS_POINTER_DEFINITION( ExplicitBuilderAndSolver );
+
+    typedef BuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef typename BaseType::TSchemeType TSchemeType;
+
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+
+    typedef typename BaseType::NodesArrayType NodesArrayType;
+
+    typedef typename BaseType::ElementsArrayType ElementsArrayType;
+
+    typedef typename BaseType::ConditionsArrayType ConditionsArrayType;
+
+
+    /*@} */
+    /**@name Life Cycle
+     */
+    /*@{ */
+
+    /** Constructor.
+     */
+    ExplicitBuilderAndSolver(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver)
+        : BuilderAndSolver< TSparseSpace, TDenseSpace, TLinearSolver >(pNewLinearSystemSolver)
+    {
+    }
+
+    /** Destructor.
+     */
+    virtual ~ExplicitBuilderAndSolver()
+    {
+    }
+
+
+    /*@} */
+    /**@name Operators
+     */
+    /*@{ */
+
+
+    /**
+     * This does not build a matrix but is used to create the LHS
+     * in explicit time integration each node is updated and thus
+     * does not need to assemble any matrix
+     */
+
+    void BuildLHS(
+            typename TSchemeType::Pointer pScheme,
+            ModelPart& rModelPart,
+            TSystemMatrixType& A)
+    {
+        KRATOS_TRY
+
+        //Set Nodal Mass to zero
+        NodesArrayType& r_nodes             = rModelPart.Nodes();
+        ElementsArrayType& r_elements       = rModelPart.Elements();
+        ProcessInfo& r_current_process_info   = rModelPart.GetProcessInfo();
+
+
+        auto it_node = rModelPart.NodesBegin();
+        #pragma omp parallel for firstprivate(it_node)
+        for(int i=0;i<static_cast<int>(r_nodes.size());++i)
+        {
+            (it_node+i)->SetValue(NODAL_MASS,0.00);
+        }
+
+        if (rModelPart.NodesBegin()->HasDofFor(ROTATION_Z))
```

Comment:
put this check out of the loop. The goal is to avoid calling many times HasDofFor